### ISSUE TITLE
Resample BM adds extra index point

### ISSUE THREAD

**Issue by bjonen** [pydata/pandas#9756]
```
In [1]: index = pd.DatetimeIndex(start='20150101', end='20150331', freq='B')

In [2]: df = pd.DataFrame(index=index, data=len(index)*[0])

In [3]: df.resample('BM', how='last')
Out[3]:
             0
2015-01-30   0
2015-02-27   0
2015-03-31   0
2015-04-30 NaN
```
This is both in 0.14.1 and 0.15.2

**Comment by rockg** [comment]
This actually might be fixed on master related to a bunch of resample fixes.

**Comment by rockg** [comment]
Confirmed that it is *not* fixed on master.

**Comment by jreback** [comment]
couple of resample bins issues still lurking; #9623 fixed a lot of them

**Comment by scari** [comment]
Working on this.

**Comment by scari** [comment]
Also reproducible /w resample BQ

**Comment by jreback** [comment]
closed by #10044
